libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2639,6 +2619,27 @@ bool TypeChecker::visit(Identifier const& _identifier)
 	else if (dynamic_cast<MagicVariableDeclaration const*>(annotation.referencedDeclaration))
 		if (dynamic_cast<FunctionType const*>(annotation.type.get()))
 			annotation.isPure = true;
+
+	// Check for deprecated function names.
+	// The check is done here for the case without an actual function call.
+	if (
```

Comment:
Actually this whole thing can be shorted to
```suggestion
	if (FunctionType const* fType = dynamic_cast<FunctionType const*>(_identifier.annotation().type))
```